scripts/application_startup/customization_dev.groovy

```diff
@@ -271,70 +271,82 @@ void fetch_files_by_hash(local_file_hash_map, remote_file_hash_map) {
+                message.add("++ Remote custom file " + remote_file_name + " has been updated, downloading now.")
+                console.println("++ Remote custom file " + remote_file_name + " has been updated, downloading now.")
+                // download
+                console.println("remote_config_dir + remote_file_name: " + remote_config_dir + File.separator + remote_file_name)
+
+                download_asset(remote_file_url)
+
+                if (remote_file_name == "omegat.prefs") {
+                    update_omegat_prefs()
+                } else if (remote_file_name == "uiLayout.xml") {
+                    update_ui_layout()
                 }
+                console.println("????????????????????????????????????")
             }
         } else {
             // just download
             // check if it exists locally, if it does, then msg:
             message.add("++ Remote custom file " + remote_file_name + " is new or had not been downloaded, downloading now.")
             console.println("++ Remote custom file " + remote_file_name + " is new or had not been downloaded, downloading now.")
             // download
-            download_asset(get_remote_file_url(remote_config_dir, remote_file_name))
+            download_asset(remote_file_url)
             // download_asset(remote_config_dir + File.separator + remote_file_name)
         }
     }
 }
 
+/** Update the UI Layout with the newly download layout. */
+// XXX This effectively prevents users to make adjustments on the UI layout as it will always be
+// reset with the remote layout at each application restart.
+void update_ui_layout() {
+    console.println("<<<< HANDLING UI LAYOUT >>>>>")
+    new File("uiLayout.xml").withInputStream(is -> {
+        Core.getMainWindow().getDesktop().readXML(is)
+    })
+}
+
 // Update current preferences from a file and save them.
-void update_omegat_prefs(localPrefsPath) {
-    // parse (now) local prefs file
-    Map<String, String> remotePrefs = get_omegat_prefs(localPrefsPath)
-    remotePrefs.each { prop -> console.println("::: remote: ${prop.key} => ${prop.value}") } // @debug
+void update_omegat_prefs() {
+    console.println("<<<< HANDLING OMEGAT PREFS >>>>>")
+
+    // parse local prefs file
+    File local_prefs_path = new File(config_dir, 'omegat.prefs') // @debug
+    //console.println('localPrefsPath  : ' + localPrefsPath) // @debug
+    //get_omegat_prefs(localPrefsPath).each { prop -> console.println(":::   local: ${prop.key} => ${prop.value}") } // @debug
+
+    // The local omegat.prefs file has been overwritten with the remote one, but the actual
+    // preferences are not yet updated!
+
+    Map<String, String> remote_prefs = get_omegat_prefs(local_prefs_path)
+    remote_prefs.each { prop -> console.println("::: remote: ${prop.key} => ${prop.value}") } // @debug
 
     console.println(" I will set scripts_dir to '${local_scripts_dpath}'")
-    remotePrefs.scripts_dir = local_scripts_dpath
```

Comment:
@briacp There's a conflict in your last PR. I'm not sure what should be merged exactly.

On the one hand, I understand you've refactored your additions to use snake_case (e.g. `remote_prefs`) as I do instead of camelCase (e..g `remotePrefs`), as you'd do. Is that correct? 

If my understanding above is correct, then on the other hand, I'm not sure whether we should use `remote_prefs.scripts_dir = local_scripts_dpath` or `remote_prefs.put('scripts_dir', local_scripts_dpath.getAbsolutePath())` to set the path of the scripts directory in the preferences. Could you please advice?

Reply:
I resolved the conflict

```java
    remote_prefs.scripts_dir = local_scripts_dpath
    remote_prefs.each { prop -> console.println("::: remote: ${prop.key} => ${prop.value}") } // @debug
```

is what I kept as the preference should be stored as a String, not a File

Reply:
It should be a String, that's why the agreed to use getAbsolutePath() - which returns a String. I propose to keep using `remote_prefs.put('scripts_dir', local_scripts_dpath.getAbsolutePath())` as in master now.

Reply:
oops you're absolutely correct, I pasted the wrong code